astropy/coordinates/matching.py

```diff
@@ -299,6 +315,14 @@ def search_around_sky(coords1, coords2, seplimit, storekdtree='_kdtree_sky'):
     if not seplimit.isscalar:
         raise ValueError('seplimit must be a scalar in search_around_sky')
 
+    if coords1.isscalar or coords2.isscalar:
+        raise ValueError('One of the inputs to search_around_sky is a scalar. '
+                         'search_around_sky is intended for use with array '
+                         'coordinates, not scalars.  Use the `separation` '
+                         'method on coordinate objects along with '
+                         '`numpy.argmin` to find the closest match to a scalar'
+                         'coordinate.')
```

Comment:
Oops, good point re: `argmin` - I was thinking about the match_to_catalog examples, and this is different.  Instead it should be just ``sep<1*u.deg`` or similar.  Will update the message appropriately.

I'll address the other item in the comments, as this thread good covered up when I push up a new commit.